spec/factories/virtual_template.rb

```diff
@@ -0,0 +1,16 @@
+FactoryGirl.define do
+  factory :virtual_template, :class => 'ManageIQ::Providers::CloudManager::VirtualTemplate' do
+    vendor 'amazon'
+    name 'vt'
+    location 'us-west-1'
+    ems_ref 'i-12345'
+    availability_zone_id 0
+    cloud_subnet_id 1
+    cloud_network_id 2
+    type 'ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate'
```

Comment:
I don't think I like the default being amazon, with generic being a trait...seems like the other way around would be more appropriate?  Additionally, it might be better to create a child factory instead of a trait since this represents a subclass (at least that's how we do it in other factories).